service/src/main/java/uk/gov/hmcts/reform/fpl/service/orders/generator/C43ChildArrangementOrderDocumentParameterGenerator.java

```diff
@@ -0,0 +1,134 @@
+        + "Passport Office, Globe House, 89 Eccleston Square, LONDON SW1V 1PN.";
+
+    private final LocalAuthorityNameLookupConfiguration laNameLookup;
+
+    @Override
+    public Order accept() {
+        return Order.C43_CHILD_ARRANGEMENT_SPECIFIC_ISSUE_PROHIBITED_STEPS_ORDER;
+    }
+
+    @Override
+    public DocmosisParameters generate(CaseData caseData) {
+        ManageOrdersEventData eventData = caseData.getManageOrdersEventData();
+
+        String localAuthorityCode = caseData.getCaseLocalAuthority();
+        String localAuthorityName = laNameLookup.getLocalAuthorityName(localAuthorityCode);
+
+        C43ChildArrangementOrderDocmosisParameters.C43ChildArrangementOrderDocmosisParametersBuilder<?, ?>
+            c43DocmosisParameters = C43ChildArrangementOrderDocmosisParameters
+            .builder()
+            .orderTitle(getOrderTitle(eventData))
+            .orderByConsent(orderMessageGenerator.getOrderByConsentMessage(eventData))
+            .orderDetails(getOrderRecitalsAndPreambles(eventData))
+            .furtherDirections(getOrderDirections(eventData))
+            .localAuthorityName(localAuthorityName)
+            .noticeHeader("Notice")
+            .noticeMessage(NOTICE);
+
+        if (showWarning(eventData)) {
+            c43DocmosisParameters
+                .orderHeader("Warning \n")
+                .orderMessage(WARNING);
+        }
+
+        return c43DocmosisParameters.build();
+    }
+
+    @Override
+    public DocmosisTemplates template() {
+        return DocmosisTemplates.ORDER_V2;
+    }
+
+    private String getOrderTitle(ManageOrdersEventData eventData) {
+        List<C43OrderType> orders = eventData.getManageOrdersC43Orders();
+
+        switch (orders.size()) {
+            case 1:
+                return String.format("%s order", orders.get(0).getLabel());
+            case 2:
+                return String.format("%s and %s order", orders.get(0).getLabel(), orders.get(1).getLabel());
+            default:
+                return String.format("%s, %s and %s order",
+                    orders.get(0).getLabel(),
+                    orders.get(1).getLabel(),
+                    orders.get(2).getLabel());
+        }
+    }
+
+    private String getOrderRecitalsAndPreambles(ManageOrdersEventData eventData) {
+        String recitals = eventData.getManageOrdersRecitalsAndPreambles();
+
+        return "The Court orders" + "\n\n" + recitals;
+    }
+
+    private String getOrderDirections(ManageOrdersEventData eventData) {
+        String directions = eventData.getManageOrdersC43Directions();
+        String furtherDirections = eventData.getManageOrdersFurtherDirections();
+
+        if (!isEmpty(furtherDirections)) {
+            directions += "\n\n" + furtherDirections;
+        }
+
+        directions += "\n\n" + WHERE;
+
+        return directions;
+    }
+
+    private Boolean showWarning(ManageOrdersEventData eventData) {
```

Comment:
Generally, Boolean variables should be prefixed with "is" or "has". The reasoning for this, is that you want a clear answer to be "Yes" or "No". So that you know what to expect.
showWarning is a verb, its a doing word/term and a clear action.

I would suggest refactoring it to be something more readable, such as 


`
 if (isChildArrangementOrderSelected(eventData)) {
            addChildArrangementOrderWarningMessage(c43DocmosisParameters);
 }
`

`
 private void addChildArrangementOrderWarningMessage(
        C43ChildArrangementOrderDocmosisParameters.C43ChildArrangementOrderDocmosisParametersBuilder<?, ?>
            c43DocmosisParameters) {
        c43DocmosisParameters
            .orderHeader("Warning \n")
            .orderMessage(WARNING);
    }
`


The best code ive read has assigned variable names so well thought out that you'd feel like a fool picking anything else. These things take time to learn but id highly recommend the book Clean Code for a bit of light reading on the topic